source/AI.cpp

```diff
@@ -115,15 +115,13 @@ namespace {
 	// Determine if the ship has any usable weapons.
 	bool IsArmed(const Ship &ship)
 	{
-		for(const Hardpoint &hardpoint : ship.Weapons())
+		for(const Hardpoint *hardpoint : ship.GetArmament().TurrettedWeaponsNoAM())
```

Comment:
You're right, thank's! ...I don't know how it could slip in :sweat: , the last commit is wrong.

I'm going to change so the forth 'category' is 'all but no AM'.